Mathlib/MeasureTheory/Integral/CircleAverage.lean

```diff
@@ -0,0 +1,133 @@
+/-
+Copyright (c) 2025 Stefan Kebekus. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stefan Kebekus
+-/
+import Mathlib.MeasureTheory.Integral.CircleIntegral
+import Mathlib.MeasureTheory.Integral.IntervalAverage
+import Mathlib.MeasureTheory.Integral.IntervalIntegral.Periodic
+
+/-!
+# Circle Averages
+
+For a function `f` on the complex plane, this file introduces the notation `circleAverage f c R` as
+a shorthand for the average of `f` on the circle with center `c` and radius `R`. Averages of this
+form are typically used in analysis of one complex variable. Like `IntervalAverage`, this notion
+exists as a convenience. It avoids the hassle to manually elemininate `2 * π` every time an average
+is computed.
+
+Note: The relevant integrability property is `CircleIntegrable`, as defined in
+`Mathlib.MeasureTheory.Integral.CircleIntegral`.
+
+Implementation Note: Like `circleMap`, `circleAverage`s are defined for negative radii. The theorem
+`circleAverage_congr_negRadius` shows that the average is independent of the radius' sign.
+-/
+
+open Filter Metric Real
+
+variable
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace ℝ E]
+  {𝕜 : Type*} [NontriviallyNormedField 𝕜] [NormedSpace 𝕜 E] [SMulCommClass ℝ 𝕜 E]
+  {f f₁ f₂ : ℂ → E} {c : ℂ} {R : ℝ} {a : 𝕜}
+
+/-!
+# Definition
+-/
+
+variable (f c R) in
+/--
+Define `circleAverage f c R` as the average value of `f` on the circle with center `c` and radius
+`R`.
+-/
+noncomputable def circleAverage : E :=
+  (2 * π)⁻¹ • ∫ θ in (0)..2 * π, f (circleMap c R θ)
+
+/-- Expression of `circleAverage´ in terms of interval averages. -/
+lemma circleAverage_eq_intervalAverage :
+    circleAverage f c R = ⨍ θ in (0)..2 * π, f (circleMap c R θ) := by
+  simp_all [circleAverage, interval_average_eq]
+
+/-- Interval averages for zero radii equal values at the center point. -/
+@[simp] lemma circleAverage_zeroRadius [CompleteSpace E] :
+    circleAverage f c 0 = f c := by
+  rw [circleAverage]
+  simp only [circleMap_zero_radius, Function.const_apply,
+    intervalIntegral.integral_const, sub_zero,
+    ← smul_assoc, smul_eq_mul, inv_mul_cancel₀ (mul_ne_zero two_ne_zero pi_ne_zero),
+    one_smul]
+
+/--
+Expression of `circleAverage´ with arbitrary center in terms of `circleAverage` with center zero.
+-/
+lemma circleAverage_shiftCenter :
+    circleAverage f c R = circleAverage (fun z ↦ f (z + c)) 0 R := by
+  unfold circleAverage circleMap
+  congr
+  ext θ
+  congr 1
+  ring
+
+/-!
+## Congruence Lemmata
+-/
+
+/-- Circle averages do not change when replacing the radius by its negative. -/
+theorem circleAverage_congr_negRadius :
```

Comment:
```suggestion
theorem circleAverage_congr_neg_radius :
```
The rule is: if it's one symbol, no underscore, and capitals. Things corresponding to different symbols are separated by underscores. Same thing below.

Reply:
Understood. Fixed as suggested.